add documentation redgtech

Jonhsady opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).

  Before submitting your pull request, please verify that you have chosen the correct target branch,
  and that the PR preview looks fine and does not include unrelated changes.
-->
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

As there hasn't been any progress in a while and the core PR has been merged in the last release, this PR is being replaced by a new PR: https://github.com/home-assistant/home-assistant.io/pull/43709


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/136947
- Link to parent pull request in the Brands repository: https://github.com/home-assistant/brands/pull/6466
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

- **New Features**
  - Added Redgtech integration for smart home devices
  - Support for controlling Redgtech smart lights and switches
  - Cloud-based device monitoring and control
  - Easy configuration through Home Assistant settings

- **Documentation**
  - Comprehensive setup guide and device compatibility information
  - Troubleshooting resources included
  - Removed outdated information regarding vehicle sleep from Teslemetry integration documentation
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Jonhsady committed:
add documentation redgtech

Jonhsady committed:
Merge branch 'current' into current

c0ffeeca7 commented:
Thank you for contributing, @Jonhsady  :+1: 

- As this PR seems to be related to a code PR, please rebase and point this PR against the `next` branch.
- The `current` branch is used for documentation that applies right *now*. 
- The documentation you are submitting here only applies once your new feature is merged, at some point in the future. And for this reason this PR needs to target the `next` branch.

c0ffeeca7 commented:
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

Jonhsady committed:
Merge branch 'next' into current

Jonhsady committed:
Merge branch 'next' into current

Jonhsady committed:
Merge branch 'next' into current

luan-nvg committed:
Merge branch 'next' into current

Jonhsady commented:
@Danielhiversen please check if everything is correct and merge thx

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,44 @@
+---
+## Redgtech
+
+The **Redgtech** integration allows you to connect compatible devices with Home Assistant, offering smart control and advanced monitoring.
```
Add a link to the redgtech website here

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,44 @@
+1. *Go to **Settings** > **Devices & Services**.*
+2. *Click on **Add Integration**.*
+3. *Choose **Redgtech** from the list.*
+4. Log in with your Redgtech app credentials.
```
Include the standard setup instruction here by `{% include integrations/config_flow.md %}`, the only non-standard part is to use the Redgtech app credentials.

emontnemery replied:
The implementation asks for "email" and "password", maybe write that out here:
```
{% configuration_basic %}
Email:
    description: "The email address connected to your account. You can find it in ..."
Passert:
    description: "The account password."
{% endconfiguration_basic %}
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,44 @@
+
+## Troubleshooting
+
+If you encounter issues using this integration, refer to the [troubleshooting guide](../troubleshooting.md) or contact our support.
```
What support? Maybe add a link to redgtech support?

emontnemery commented on source/_integrations/redgtech.markdown:
Removal instructions are required, you can just include the blurb from here: https://developers.home-assistant.io/docs/core/integration-quality-scale/rules/docs-removal-instructions#example-implementation

Jonhsady committed:
docs: add missing installation and removal instructions for redgtech

- Add ## Installation section with prerequisites
- Add ## Removal section with step-by-step instructions
- Complete docs-installation-instructions requirement
- Complete docs-removal-instructions requirement
- Address @emontnemery feedback on missing docs sections

Jonhsady committed:
docs: complete redgtech integration documentation

- Add link to Redgtech website (https://redgtech.com.br/)
- Include standard config_flow.md template
- Add configuration_basic section with Email and Password fields
- Implement standard removal instructions with remove_device_service.md
- Add Redgtech app removal step
- Update support contact to contato.redgtech.dev@gmail.com
- Complete all required documentation sections for quality scale

Jonhsady committed:
docs: add installation parameters section for redgtech integration

- Add Installation Parameters section documenting no special requirements
- Clarify cloud-based communication approach
- List what is NOT required (local network config, port forwarding, etc.)
- Complete docs-installation-parameters requirement per @emontnemery feedback

Jonhsady committed:
docs: add automation examples and use cases for redgtech integration

- Add comprehensive automation examples section with practical use cases
- Include morning routine, evening routine, and away mode automations
- Add use cases section covering smart home automation, energy management, and convenience
- Complete docs-examples and docs-use-cases requirements per quality scale
- Provide real-world automation scenarios users can implement

Jonhsady committed:
docs: add data updates and known limitations sections for redgtech integration

- Add comprehensive Data Updates section explaining polling approach and data flow
- Document update interval (60 seconds), authentication handling, and retry logic
- Add Known Limitations section covering cloud dependency, polling interval, and API limits
- Complete docs-data-update and docs-known-limitations requirements per quality scale
- Provide detailed technical information about how the integration works

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,210 @@
+- Your Redgtech devices connected to your network
+- Internet connectivity for cloud communication
+
+### Installation Parameters
```
```suggestion
### Installation parameters
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization), also in headings. Please fix throughout the documentation

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,210 @@
+
+{% include integrations/remove_device_service.md %}
+
+After deleting the integration, go to the Redgtech app and remove the Home Assistant integration from there as well.
```
```suggestion
```
This section is typically at the very end of the page, as it marks the end of the integration life cycle. For more information, refer to the documentation on [Integration documentation structure](https://developers.home-assistant.io/docs/documenting/integration-docs-examples). It shows one example of a structure. [Some other documentation examples](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown)

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,210 @@
+2. Check if your Redgtech account is active
+3. Try logging into the Redgtech app to confirm your credentials
+
+For additional support, please contact [Redgtech support](mailto:contato.redgtech.dev@gmail.com) directly.
```
```suggestion
For additional support, please contact [Redgtech support](mailto:contato.redgtech.dev@gmail.com) directly.

## Removing the integration

This integration follows standard integration removal. No extra steps are required.

{% include integrations/remove_device_service.md %}

After deleting the integration, go to the Redgtech app and remove the Home Assistant integration from there as well.
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,210 @@
+
+If you receive authentication errors:
+
+1. Verify your email and password are correct
```
```suggestion
1. Verify your email and password are correct.
```
End sentences with periods.

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,210 @@
+## Use Cases
+
+### Smart Home Automation
+- **Morning Routine**: Automatically turn on bedroom lights when your alarm goes off
```
```suggestion
- **Morning routine**: Automatically turn on bedroom lights when your alarm goes off.
```
We use sentence-style capitalization also for UI strings. Please end sentences with periods.

Jonhsady committed:
Apply Microsoft style capitalization: sentence case for all headings and UI strings, add periods to lists

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+  - '@luan-nvg'
+---
+
+The **Redgtech** integration allows you to control your Redgtech smart devices through Home Assistant. This integration provides seamless connectivity to your Redgtech ecosystem, enabling you to manage lights and switches from within your Home Assistant dashboard.
```
```suggestion
The **Redgtech** integration allows you to control your [Redgtech](https://redgtech.com.br/) smart devices through Home Assistant. This integration provides seamless connectivity to your Redgtech ecosystem, enabling you to manage lights and switches from within your Home Assistant dashboard.
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+
+The **Redgtech** integration allows you to control your Redgtech smart devices through Home Assistant. This integration provides seamless connectivity to your Redgtech ecosystem, enabling you to manage lights and switches from within your Home Assistant dashboard.
+
+For more information about Redgtech, visit the [Redgtech website](https://redgtech.com.br/).
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+## Installation
+
+The Redgtech integration is available in Home Assistant by default. No additional installation steps are required.
+
```
```suggestion

```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+
+The Redgtech integration is available in Home Assistant by default. No additional installation steps are required.
+
+### Prerequisites
```
```suggestion
## Prerequisites
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+- Your Redgtech devices connected to your network.
+- Internet connectivity for cloud communication.
+
+### Installation parameters
```
```suggestion
## Installation parameters
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,211 @@
+
+All communication is handled through the Redgtech cloud service using standard HTTPS connections.
+
+## Configuration
```
```suggestion
```
This title is not needed here. It is already part of the include section.
<img width="861" height="569" alt="image" src="https://github.com/user-attachments/assets/7cbb9c59-116f-4cd2-a9d8-6e22639a1928" />

Jonhsady committed:
Update Redgtech integration documentation

- Add link to Redgtech website in introduction
- Remove redundant installation information
- Update section headers to use ## instead of ###
- Remove Configuration section header

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+title: Redgtech
+description: Integrate Redgtech smart devices with Home Assistant
+ha_category:
+  - Light
```
The initial core PR only has switch. We should not add light here yet.

c0ffeeca7 replied:
```suggestion
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+
+## Installation parameters
+
+The Redgtech integration does not require any special installation parameters. The integration uses cloud-based communication and does not require:
+
+- Local network configuration
+- Port forwarding
+- Static IP addresses
+- Special firewall rules
+
+All communication is handled through the Redgtech cloud service using standard HTTPS connections.
```
There's no need to list what we don't need.

```suggestion
The Redgtech integration does not require any special installation parameters other than Redgtech account email and password.

All communication is handled through the Redgtech cloud service using standard HTTPS connections.
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+## Features
+
+- **Smart switch control**: Turn your Redgtech switches on/off remotely.
+- **Smart light control**: Control brightness and on/off state of Redgtech lights.
```
Remove this for now

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+- **Smart switch control**: Turn your Redgtech switches on/off remotely.
+- **Smart light control**: Control brightness and on/off state of Redgtech lights.
+- **Real-time status**: See the current state of your devices in real time.
+- **Automation support**: Use Redgtech devices in Home Assistant automations.
```
All Home Assistant integration allow automating, it's not a special feature of this integration

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+- **Smart light control**: Control brightness and on/off state of Redgtech lights.
+- **Real-time status**: See the current state of your devices in real time.
+- **Automation support**: Use Redgtech devices in Home Assistant automations.
+- **Cloud connectivity**: Access your devices from anywhere via Redgtech cloud.
```
Remove, it's irrelevant for the Home Assistant ingegration

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+| Device Type | Models | Features |
+|-------------|--------|----------|
+| Smart Switch | X1, X2, X3 | On/Off control, Status monitoring |
+| Smart Light | B1, B2, B3 | On/Off control, Brightness control |
```
Remove the smart light from here for now

c0ffeeca7 replied:
```suggestion
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+
+| Device Type | Models | Features |
+|-------------|--------|----------|
+| Smart Switch | X1, X2, X3 | On/Off control, Status monitoring |
```
What is X1, X2, X3? 
What do you mean by "status monitoring"?

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+
+The Redgtech integration uses a polling-based approach to keep device data synchronized:
+
+- **Update interval**: Device states are updated every 60 seconds.
```
The time is not correct, in the code it's 15 seconds

c0ffeeca7 replied:
```suggestion
- **Update interval**: Device states are updated every 15 seconds.
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+| Smart Switch | X1, X2, X3 | On/Off control, Status monitoring |
+| Smart Light | B1, B2, B3 | On/Off control, Brightness control |
+
+## Data updates
+
+The Redgtech integration uses a polling-based approach to keep device data synchronized:
+
+- **Update interval**: Device states are updated every 60 seconds.
+- **Authentication**: The integration automatically handles token renewal when needed.
+- **Error handling**: If authentication fails, the integration will trigger a reauthentication flow.
+- **Retry logic**: Temporary connection issues are handled with automatic retries.
+- **Real-time control**: Device commands (on/off, brightness) are sent immediately without waiting for the next update cycle.
+
+### Data flow
+
+1. **Initial setup**: When you first configure the integration, it authenticates with the Redgtech API.
+2. **Token management**: The integration stores an access token and automatically renews it when needed.
+3. **Device discovery**: All available devices are fetched from your Redgtech account.
+4. **Periodic updates**: Every 60 seconds, the integration polls the API for the latest device states.
+5. **Command execution**: When you control a device, the command is sent immediately to the API.
+6. **State synchronization**: Device states are updated in Home Assistant to reflect the current status.
```
I don't understand what these sections add, I'd suggest to remove them

c0ffeeca7 replied:
```suggestion.
```

emontnemery commented on source/_integrations/redgtech.markdown:
````diff
@@ -0,0 +1,204 @@
+```
+
+### Turn off all Redgtech switches when leaving home
+
+```yaml
+automation:
+  - alias: "Turn off Redgtech switches when leaving"
+    trigger:
+      platform: state
+      entity_id: person.your_name
+      to: "not_home"
+    action:
+      service: switch.turn_off
+      target:
+        entity_id: switch.redgtech_*
+```
+
+### Schedule Redgtech devices
+
+```yaml
+automation:
+  - alias: "Morning routine - Turn on bedroom light"
+    trigger:
+      platform: time
+      at: "07:00:00"
+    condition:
+      condition: time
+      weekday:
+        - mon
+        - tue
+        - wed
+        - thu
+        - fri
+    action:
+      service: light.turn_on
+      target:
+        entity_id: light.redgtech_bedroom_light
+      data:
+        brightness: 255
+```
````
Please remove this. Automation examples are relevant for integrations which implement custom actions, your integration doesn't.

c0ffeeca7 replied:
```suggestion
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+
+- **Smart switch control**: Turn your Redgtech switches on/off remotely.
+- **Smart light control**: Control brightness and on/off state of Redgtech lights.
+- **Real-time status**: See the current state of your devices in real time.
```
This is not true since the integration polls the cloud every 15 seconds

emontnemery commented on source/_integrations/redgtech.markdown:
````diff
@@ -0,0 +1,204 @@
+        brightness: 255
+```
+
+## Use cases
+
+### Smart home automation
+- **Morning routine**: Automatically turn on bedroom lights when your alarm goes off.
+- **Evening routine**: Gradually dim lights as bedtime approaches.
+- **Away mode**: Turn off all Redgtech devices when you leave home.
+- **Security**: Turn on lights when motion is detected in specific areas.
+
+### Energy management
+- **Scheduled control**: Set timers to automatically turn off devices after a certain period.
+- **Occupancy-based control**: Turn devices on/off based on room occupancy.
+- **Sunset/sunrise**: Sync device behavior with natural light patterns.
+
+### Convenience
+- **Voice control**: Use voice assistants to control Redgtech devices.
+- **Remote access**: Control devices from anywhere using the Home Assistant mobile app.
+- **Scene control**: Create scenes that control multiple Redgtech devices simultaneously.
````
I'm not sure I see the need for this list of use cases. Any integration which have switch entities supports the same use cases, it's not unique for redgtech.

c0ffeeca7 replied:
```suggestion
```

emontnemery commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+- **Remote access**: Control devices from anywhere using the Home Assistant mobile app.
+- **Scene control**: Create scenes that control multiple Redgtech devices simultaneously.
+
+## Known limitations
+
+- **Cloud dependency**: The integration requires internet connectivity to function.
+- **Polling interval**: Device states are updated every 60 seconds, not in real time.
+- **Single account**: Each integration instance can only connect to one Redgtech account.
+- **No local control**: Devices cannot be controlled locally without internet connection.
+- **API rate limits**: The Redgtech API has rate limits that may affect frequent updates.
```
How is the user supposed to use this list of limitations? Apart from the "single account" - which I don't even understand why you list as a limitation - the other 4 items are caused by this integration being `cloud_polling`, and IMO there's no need to list those.

c0ffeeca7 replied:
```suggestion
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+---
+
+The **Redgtech** integration allows you to control your [Redgtech](https://redgtech.com.br/) smart devices through Home Assistant. This integration provides seamless connectivity to your Redgtech ecosystem, enabling you to manage lights and switches from within your Home Assistant dashboard.
+
+
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+## Installation parameters
+
+The Redgtech integration does not require any special installation parameters. The integration uses cloud-based communication and does not require:
+
+- Local network configuration
+- Port forwarding
+- Static IP addresses
+- Special firewall rules
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+Password:
+  description: "The account password."
+{% endconfiguration_basic %}
+
```
```suggestion
```

c0ffeeca7 commented on source/_integrations/redgtech.markdown:
```diff
@@ -0,0 +1,204 @@
+The Redgtech integration uses a polling-based approach to keep device data synchronized:
+
+- **Update interval**: Device states are updated every 60 seconds.
+- **Authentication**: The integration automatically handles token renewal when needed.
+- **Error handling**: If authentication fails, the integration will trigger a reauthentication flow.
+- **Retry logic**: Temporary connection issues are handled with automatic retries.
+- **Real-time control**: Device commands (on/off, brightness) are sent immediately without waiting for the next update cycle.
```
```suggestion
- Authentication: The integration automatically handles token renewal when needed.
- Error handling: If authentication fails, the integration will trigger a reauthentication flow.
- Retry logic: Temporary connection issues are handled with automatic retries.
- Real-time control: Device commands (on/off, brightness) are sent immediately without waiting for the next update cycle.
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide)
